lib/eBPF.mli

```diff
@@ -9,9 +9,83 @@ type size =
 | H (** half-word = 16 bit *)
 | B (** byte *)
 | DW (** double word = 64 bit *)
-
-type reg = R0 | R1 | R2 | R3 | R4 | R5 | R6 | R7 | R8 | R9 | R10
+[@@deriving enum]
+
+type reg =
+  | R0
+  | R1
+  | R2
+  | R3
+  | R4
+  | R5
+  | R6
+  | R7
+  | R8
+  | R9
+  | R10
+[@@deriving enum]
+
+type op_alu =
+  | ADD
+  | SUB
+  | MUL
+  | DIV
+  | OR
+  | AND
+  | LSH
+  | RSH
+  | NEG
+  | MOD
+  | XOR
+  | MOV
+  | ARSH
+  | END
+[@@deriving enum]
+
+type op_jmp =
+  | JA
+  | JEQ
+  | JGT
+  | JGE
+  | JSET
+  | JNE
+  | JSGT
+  | JSGE
+  | CALL
+  | EXIT
+  | JLT
+  | JLE
+  | JSLT
+  | JSLE
+[@@deriving enum]
+
+type source = SRC_IMM | SRC_REG [@@deriving enum]
+
+type mode =
+  | IMM
+  | ABS_todo
+  | IND_todo
+  | MEM
+  | LEN_reserved
+  | MSH_reserved
+  | XADD_todo
+[@@deriving enum]
+  
+type op =
+  | LD of size * mode | LDX of size * mode | ST of size * mode | STX of size * mode
+  | ALU of source * op_alu
+  | ALU64 of source * op_alu
+  | JMP of source * op_jmp
+  | JMP32 of source * op_jmp
 type int16 = int
+type ('op, 'reg) insn_t = {
```

Comment:
i am not sure i want to expose implementation of these types, but i understand it is needed for the parser. Will see later, there must be a way to make it visible inside library but not outside (private_modules in dune iirc)